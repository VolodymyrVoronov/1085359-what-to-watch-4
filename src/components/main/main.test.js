import React from "react";
import renderer from "react-test-renderer";
import Main from "./main.jsx";

const CURRENT_MOVIE = {
  id: 0,
  title: `Citizen Kane`,
  genre: `Drama`,
  img: `img/bg-the-grand-budapest-hotel.jpg`,
  releaseDate: 2014,

  poster: `img/bg-the-grand-budapest-hotel.jpg`,
  rating: 7.8,
  ratingReviewsCount: 10,
  description: `Lorem ipsum dolor sit amet consectetur adipisicing elit. Reiciendis natus ipsa ut     possimus laudantium excepturi magni! Soluta doloribus facere quaerat, optio ab ratione quas provident nobis. Accusamus incidunt unde dicta.`,
  story: `Lorem ipsum dolor sit amet consectetur adipisicing elit. Reiciendis natus ipsa ut     possimus laudantium excepturi magni! Soluta doloribus facere quaerat, optio ab ratione quas provident nobis. Accusamus incidunt unde dicta.`,
  director: `Director`,
  actors: [
    `Actor 1`,
    `Actor 2`,
    `Actor 3`,
    `Actor 4`,
    `Actor 5`,
  ],
  runTime: 200,
};

const MOVIES = [
  {
    id: 1,
    title: `Citizen Kane`,
    genre: `Drama`,
    img: `img/bg-the-grand-budapest-hotel.jpg`,
    releaseDate: 2014,

    poster: `img/bg-the-grand-budapest-hotel.jpg`,
    rating: 7.8,
    ratingReviewsCount: 10,
    description: `Lorem ipsum dolor sit amet consectetur adipisicing elit. Reiciendis natus ipsa ut     possimus laudantium excepturi magni! Soluta doloribus facere quaerat, optio ab ratione quas provident nobis. Accusamus incidunt unde dicta.`,
    story: `Lorem ipsum dolor sit amet consectetur adipisicing elit. Reiciendis natus ipsa ut     possimus laudantium excepturi magni! Soluta doloribus facere quaerat, optio ab ratione quas provident nobis. Accusamus incidunt unde dicta.`,
    director: `Director`,
    actors: [
      `Actor 1`,
      `Actor 2`,
      `Actor 3`,
      `Actor 4`,
      `Actor 5`,
    ],
    runTime: 200,
  },

  {
    id: 2,
    title: `Casablanka`,
    genre: `Drama`,
    img: `img/bg-the-grand-budapest-hotel.jpg`,
    releaseDate: 2014,

    poster: `img/bg-the-grand-budapest-hotel.jpg`,
    rating: 7.8,
    ratingReviewsCount: 10,
    description: `Lorem ipsum dolor sit amet consectetur adipisicing elit. Reiciendis natus ipsa ut     possimus laudantium excepturi magni! Soluta doloribus facere quaerat, optio ab ratione quas provident nobis. Accusamus incidunt unde dicta.`,
    story: `Lorem ipsum dolor sit amet consectetur adipisicing elit. Reiciendis natus ipsa ut     possimus laudantium excepturi magni! Soluta doloribus facere quaerat, optio ab ratione quas provident nobis. Accusamus incidunt unde dicta.`,
    director: `Director`,
    actors: [
      `Actor 1`,
      `Actor 2`,
      `Actor 3`,
      `Actor 4`,
      `Actor 5`,
    ],
    runTime: 200,
  },

  {
    id: 3,
    title: `The Godfather`,
    genre: `Drama`,
    img: `img/bg-the-grand-budapest-hotel.jpg`,
    releaseDate: 2014,

    poster: `img/bg-the-grand-budapest-hotel.jpg`,
    rating: 7.8,
    ratingReviewsCount: 10,
    description: `Lorem ipsum dolor sit amet consectetur adipisicing elit. Reiciendis natus ipsa ut     possimus laudantium excepturi magni! Soluta doloribus facere quaerat, optio ab ratione quas provident nobis. Accusamus incidunt unde dicta.`,
    story: `Lorem ipsum dolor sit amet consectetur adipisicing elit. Reiciendis natus ipsa ut     possimus laudantium excepturi magni! Soluta doloribus facere quaerat, optio ab ratione quas provident nobis. Accusamus incidunt unde dicta.`,
    director: `Director`,
    actors: [
      `Actor 1`,
      `Actor 2`,
      `Actor 3`,
      `Actor 4`,
      `Actor 5`,
    ],
    runTime: 200,
  },
];

const HANDLE_CLICK = () => {};

it(`<Should Main render correctly`, () => {
  const result = renderer
    .create(<Main
      currentMovie={CURRENT_MOVIE}
      films={MOVIES}
      onTitleClickHandler={HANDLE_CLICK}
    />)
    .toJSON();

  expect(result).toMatchSnapshot();
});
