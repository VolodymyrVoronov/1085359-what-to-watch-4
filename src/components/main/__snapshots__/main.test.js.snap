// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Should Main render correctly 1`] = `
Array [
  <section
    className="movie-card"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="The Grand Budapest Hotel"
        src="img/bg-the-grand-budapest-hotel.jpg"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header movie-card__head"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="user-block"
      >
        <div
          className="user-block__avatar"
        >
          <img
            alt="User avatar"
            height="63"
            src="img/avatar.jpg"
            width="63"
          />
        </div>
      </div>
    </header>
    <div
      className="movie-card__wrap"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster"
        >
          <img
            alt="The Grand Budapest Hotel poster"
            height="327"
            src="img/the-grand-budapest-hotel-poster.jpg"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            The Grand Budapest Hotel
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Drama
            </span>
            <span
              className="movie-card__year"
            >
              2014
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <button
              className="btn btn--play movie-card__button"
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </button>
            <button
              className="btn btn--list movie-card__button"
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlinkHref="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog"
    >
      <h2
        className="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <ul
        className="catalog__genres-list"
      >
        <li
          className="catalog__genres-item catalog__genres-item--active"
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            All genres
          </a>
        </li>
        <li
          className="catalog__genres-item"
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            Comedies
          </a>
        </li>
        <li
          className="catalog__genres-item"
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            Crime
          </a>
        </li>
        <li
          className="catalog__genres-item"
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            Documentary
          </a>
        </li>
        <li
          className="catalog__genres-item"
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            Dramas
          </a>
        </li>
        <li
          className="catalog__genres-item"
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            Horror
          </a>
        </li>
        <li
          className="catalog__genres-item"
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            Kids & Family
          </a>
        </li>
        <li
          className="catalog__genres-item"
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            Romance
          </a>
        </li>
        <li
          className="catalog__genres-item"
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            Sci-Fi
          </a>
        </li>
        <li
          className="catalog__genres-item"
        >
          <a
            className="catalog__genres-link"
            href="#"
          >
            Thrillers
          </a>
        </li>
      </ul>
      <div
        className="catalog__movies-list"
      >
        <article
          className="small-movie-card catalog__movies-card"
        >
          <div
            className="small-movie-card__image"
          >
            <img
              alt="Citizen Kane"
              height="175"
              src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
              onClick={[Function]}
            >
              Citizen Kane
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
        >
          <div
            className="small-movie-card__image"
          >
            <img
              alt="Casablanka"
              height="175"
              src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
              onClick={[Function]}
            >
              Casablanka
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
        >
          <div
            className="small-movie-card__image"
          >
            <img
              alt="The Godfather"
              height="175"
              src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
              onClick={[Function]}
            >
              The Godfather
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
        >
          <div
            className="small-movie-card__image"
          >
            <img
              alt="Gone With The Wind"
              height="175"
              src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
              onClick={[Function]}
            >
              Gone With The Wind
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
        >
          <div
            className="small-movie-card__image"
          >
            <img
              alt="Lawrence Of Arabia"
              height="175"
              src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
              onClick={[Function]}
            >
              Lawrence Of Arabia
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
        >
          <div
            className="small-movie-card__image"
          >
            <img
              alt="The Wizard Of Oz"
              height="175"
              src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
              onClick={[Function]}
            >
              The Wizard Of Oz
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
        >
          <div
            className="small-movie-card__image"
          >
            <img
              alt="The Graduate"
              height="175"
              src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
              onClick={[Function]}
            >
              The Graduate
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
        >
          <div
            className="small-movie-card__image"
          >
            <img
              alt="On The Waterfront"
              height="175"
              src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
              onClick={[Function]}
            >
              On The Waterfront
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
        >
          <div
            className="small-movie-card__image"
          >
            <img
              alt="Schindler's List"
              height="175"
              src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
              onClick={[Function]}
            >
              Schindler's List
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
        >
          <div
            className="small-movie-card__image"
          >
            <img
              alt="Singing' In The Rain"
              height="175"
              src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
              onClick={[Function]}
            >
              Singing' In The Rain
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
        >
          <div
            className="small-movie-card__image"
          >
            <img
              alt="It's A Wonderful Life"
              height="175"
              src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
              onClick={[Function]}
            >
              It's A Wonderful Life
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
        >
          <div
            className="small-movie-card__image"
          >
            <img
              alt="Sunset Blvd."
              height="175"
              src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
              onClick={[Function]}
            >
              Sunset Blvd.
            </a>
          </h3>
        </article>
        ;
      </div>
      <div
        className="catalog__more"
      >
        <button
          className="catalog__button"
          type="button"
        >
          Show more
        </button>
      </div>
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          © 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
